src/Element/EntityAutocompleteMultiple.php

```diff
@@ -0,0 +1,277 @@
+   *   The form state.
+   * @param array $complete_form
+   *   The original complete form array.
+   */
+  public static function validateEntityAutocompleteItems(array &$element, FormStateInterface $form_state, array &$complete_form): void {
+    // We do not need to use #value element for avoiding double work
+    // related to handling values for entity_autocomplete form elements.
+    // Instead, we can just slightly cleanup and extract needed data from
+    // ready form state values.
+    $values = $form_state->getValue(implode('.', $element['#parents'])) ?? [];
+    unset($values['add_more']);
+    $values = array_filter($values, function ($value) {
+      return !empty($value['target']);
+    });
+    usort($values, function ($a, $b) {
+      return SortArray::sortByKeyInt($a, $b, '_weight');
+    });
+    $entity_ids = array_map(function (array $item) {
+      return $item['target'];
+    }, $values);
+
+    if (empty($entity_ids)) {
+      $form_state->setError($element, t('You have to select at least 1 entity reference.'));
+    }
+
+    $form_state->setValueForElement($element, $entity_ids);
+
+  }
+
+  /**
+   * Handles the "Add another item" button AJAX request.
+   *
+   * @param array $form
+   *   The build form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public static function addMoreSubmit(array $form, FormStateInterface $form_state): void {
+    $button = $form_state->getTriggeringElement();
+
+    // Go one level up in the form, to the widgets container.
+    $element = NestedArray::getValue($form, array_slice($button['#array_parents'], 0, -1));
+    $field_name = $element['#field_name'];
+    $parents = $element['#parents'];
+
+    // Increment the items count.
+    $field_state = static::getWidgetState($parents, $field_name, $form_state);
+    $field_state['items_count']++;
+    static::setWidgetState($parents, $field_name, $form_state, $field_state);
+
+    $form_state->setRebuild();
+  }
+
+  /**
+   * Ajax callback for the "Add another item" button.
+   *
+   * @param array $form
+   *   The build form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   *
+   * @return array|null
+   *   The element.
+   */
+  public static function addMoreAjax(array $form, FormStateInterface $form_state): ?array {
+    $button = $form_state->getTriggeringElement();
+    return NestedArray::getValue($form, array_slice($button['#array_parents'], 0, -1));
+  }
+
+  /**
+   * Helper method for extracting form element state data.
+   *
+   * @param array $parents
+   *   The parents array.
+   * @param string $field_name
+   *   The field name.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   *
+   * @return array|null
+   *   Actual form element state.
+   */
+  public static function getWidgetState(array $parents, string $field_name, FormStateInterface $form_state): ?array {
```

Comment:
These methods are all direct copies from WidgetBase. Let's add a `@see` reference here.

Reply:
Done also for other copied methods.